src/main/java/org/advent/dayeight/DayEight.java

```
@@ -0,0 +1,119 @@
+            for (int i = 2; i < numberOfStepsList.size(); i++) {
+                lcmOfAllValues = lcm(lcmOfAllValues, numberOfStepsList.get(i));
+            }
+        }
+        return lcmOfAllValues;
+    }
+
+    @Override
+    public void printPartOne() {
+        System.out.println(" ");
+        System.out.println(this.getClass().getSimpleName() + " ---------------------------");
+        System.out.println("Total number of steps for part one = " + results(true));
+    }
+
+    @Override
+    public void printPartTwo() {
+        System.out.println(" ");
+        System.out.println(this.getClass().getSimpleName() + " ---------------------------");
+        System.out.println("Total number of steps for part two = " + results(false));
+    }
+
+    private long numberOfStepsToFindString(List<String> listOfStrings, Node currentNode, String directions, String stringToFind) {
+        int numberOfSteps = 0;
+        for (int i = 0; i < directions.length(); i++) {
+            if (currentNode.address().endsWith(stringToFind)) {
+                break;
+            }
+            currentNode = findNodeMapFromPointer(nodeMap(listOfStrings), currentNode, directions.charAt(i));
+            if (i == directions.length() - 1) {
+                i = -1;
+            }
+            numberOfSteps += 1;
+        }
+        return numberOfSteps;
+    }
+
+    private Map<String, Node> getFilteredMap(Map<String, Node> nodeMap) {
+        return nodeMap.entrySet().stream().filter(entry -> entry.getValue().address().endsWith("A")).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private Long results(boolean isPartOne) {
+        List<String> allStrings;
+        String directions;
+        try {
+            allStrings = Util.readStringsFromFile(Paths.get("src/main/resources/dayEight.txt").toString());
+            directions = allStrings.get(0);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
```

you shouldnt mask exceptions